apis/v1alpha1/snippetsfilter_types.go

```diff
@@ -0,0 +1,105 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:categories=nginx-gateway-fabric,shortName=snippetsfilter
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+
+// SnippetsFilter is a filter that allows inserting NGINX configuration into the
+// generated NGINX config for HTTPRoute and GRPCRoute resources.
+type SnippetsFilter struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the SnippetsFilter.
+	Spec SnippetsFilterSpec `json:"spec"`
+
+	// Status defines the state of the SnippetsFilter.
+	Status SnippetsFilterStatus `json:"status,omitempty"`
+}
+
+// SnippetsFilterSpec defines the desired state of the SnippetsFilter.
+type SnippetsFilterSpec struct {
+	// Snippets is a list of NGINX configuration snippets.
+	// There can only be one snippet per context.
+	// Allowed contexts: http, http.server, http.server.location.
+	Snippets []Snippet `json:"snippets"`
+}
+
+// Snippet represents an NGINX configuration snippet.
+type Snippet struct {
+	// Context is the NGINX context to insert the snippet into.
+	Context NginxContext `json:"context"`
+
+	// Value is the NGINX configuration snippet.
+	Value string `json:"value"`
+}
+
+// SnippetsFilterList contains a list of SnippetFilters.
+type SnippetsFilterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []SnippetsFilter `json:"items"`
+}
```

Comment:
For consistency, can you move this right underneath SnippetsFilter above? (similar to other policy definition files)

Reply:
Also, you're missing `// +kubebuilder:object:root=true` above the List definition. See other policies, we have to define this annotation on any top level struct, so we'll have it in here twice.